From 08a6d1c07ca7b4001ac3b8c579b60308e96d0b60 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Wed, 12 Jun 2024 15:09:42 +0200
Subject: [PATCH 645/954] fix(stm32mp1-ssp): fix minor issue in TFA SSP
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

NOT UPSTREAMABLE

- Remove "stm32mp_print_cpuinfo" because it is already called before
fix
- correct an issue with otp cleanup
- clarify the code when needed

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d344df3c4afd54d57ed5f06b03d8f04dff0b84ff fix(stm32mp1-ssp): fix minor issue in TFA SSP]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: I1df2720f0f32f367acff2e5eb27852181cb945dd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/387259
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp1/stm32mp1_ssp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plat/st/stm32mp1/stm32mp1_ssp.c b/plat/st/stm32mp1/stm32mp1_ssp.c
index 2819242b8a..e75cacba09 100644
--- a/plat/st/stm32mp1/stm32mp1_ssp.c
+++ b/plat/st/stm32mp1/stm32mp1_ssp.c
@@ -152,11 +152,12 @@ static int ssp_pub_key_prog(boot_api_context_t *boot_context)
 			return -EINVAL;
 		}
 
-		value++;
 		if (bsec_permanent_lock_otp(i) != BSEC_OK) {
 			ERROR("Error locking OTP %u\n", i);
 			panic();
 		}
+
+		value++;
 	}
 
 	return 0;
@@ -502,7 +503,7 @@ static void cleanup_otp_value(void)
 	}
 
 	for (i = SSP_OTP_SECRET_BASE; i < SSP_OTP_SECRET_END + 1U; i++) {
-		if (bsec_write_otp(0U, otp_rma.idx) != BSEC_OK) {
+		if (bsec_write_otp(0U, i) != BSEC_OK) {
 			return;
 		}
 	}
@@ -1015,7 +1016,6 @@ void bl2_el3_plat_arch_setup(void)
 	if ((boot_context->p_ssp_config == NULL) ||
 	    (boot_context->p_ssp_config->ssp_cmd !=
 	     BOOT_API_CTX_SSP_CMD_PROV_SECRET_ACK)) {
-		stm32mp_print_cpuinfo();
 		if (!stm32mp_is_auth_supported()) {
 			ERROR("Chip doesn't support SSP\n");
 			panic();
-- 
2.53.0

